src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -563,7 +573,119 @@ public static void renderTileItem(Item item, int metadata)
     @Deprecated
     public static void registerTESRItemStack(Item item, int metadata, Class<? extends TileEntity> TileClass)
     {
-        tileItemMap.put(Pair.of(item, metadata), TileClass);
+    	tileItemMap.put(Pair.of(item, metadata), TileClass);
+    }
+
+    public static void renderLitItem(RenderItem ri, IBakedModel model, int color, ItemStack stack) 
+    {
+        BufferBuilder bufferbuilder = Tessellator.getInstance().getBuffer();
+
+        List<BakedQuad> allquads = new ArrayList<>();
+
+        for (EnumFacing enumfacing : EnumFacing.values())
+        {
+            allquads.addAll(model.getQuads(null, enumfacing, 0));
+        }
+
+        allquads.addAll(model.getQuads(null, null, 0));
+
+        // Current list of consecutive quads with the same lighting
+        List<BakedQuad> segment = new ArrayList<>();
+
+        // Lighting of the current segment
+        int segmentBlockLight = -1;
+        int segmentSkyLight = -1;
+
+        for (int i = 0; i < allquads.size(); i++) {
+            BakedQuad q = allquads.get(i);
+
+            // Light value of current quad
+            int[] light = { 0, 0 };
+
+            // Fail-fast on ITEM, as it cannot have light data
+            // Otherwise, inspect the format for TEX_2S (lightmap)
+            if (q.getFormat() != DefaultVertexFormats.ITEM && q.getFormat().getElements().contains(DefaultVertexFormats.TEX_2S)) {
```

Comment:
The `contains` here can be `indexOf(...) != -1`, and then that doesn't need redoing next line.